yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -0,0 +1,267 @@
+from yo1k.tic_tac_toe.core.game import (
+    ActionQueue,
+    Action,
+    Phase,
+    Player,
+    State,
+    Logic,
+    Board,
+    Cell,
+    Mark)
+from collections.abc import Sequence
+from typing import Optional
+import unittest
+
+
+class ListActionQueue(ActionQueue):
+    def __init__(self, actions: Sequence[Optional[Action]]):
+        self.actions = actions
+        self.i = 0
+
+    def next(self) -> Optional[Action]:
+        if self.i >= len(self.actions):
+            return None
+        action = self.actions[self.i]
+        self.i += 1
+        return action
+
+
+def _new_state(
+        game_rounds=5,
+        player1=None,
+        player2=None,
+        board=None,
+        phase=Phase.INROUND,
+        round_=0,
+        step=0) -> State:
+    player1 = Player(Mark.X) if player1 is None else player1
+    player2 = Player(Mark.O) if player2 is None else player2
```

Comment:
Rename `player1`/`player2` to `player_x`/`player_o`. Otherwise, a caller may pass `Player(Mark.O)` as `player1` and `None` as `player2` without suspecting anything, which results in having two players with the `O` mark.

Reply:
Done.